pkg/eks/podidentities/procedures.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podidentities
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/eks/eksiface"
+)
+
+var (
+	// ErrNilPodIdentityAssociation defines an error for when a nil eks pod identity is returned.
+	ErrNilPodIdentityAssociation = errors.New("nil eks pod identity association returned from create")
+	// ErrPodIdentityAssociationNotFound defines an error for when an eks pod identity is not found.
+	ErrPodIdentityAssociationNotFound = errors.New("eks pod identity association not found")
+	// ErrPodIdentityAssociationAlreadyExists defines an error for when an eks pod identity already exists.
+	ErrPodIdentityAssociationAlreadyExists = errors.New("eks pod identity association already exists")
+)
```

Comment:
Do these need to be globally defined? These become part of the interface and users may expect them to be here and use them (but shouldn't).

Reply:
I will admit that I copied this type of implementation from `addons.go` and chose consistency over preference. Ill unexport them as I do agree that there is no current usecase for exporting them.